// @voidsolutions/vue-dateformat/dist/vue-dateformat.es.js
import { computed as a, openBlock as r, createElementBlock as u, normalizeClass as c, unref as o, toDisplayString as i } from "vue";
const m = {
  __name: "DateFormat",
  props: {
    date: {
      type: [String, Date],
      default: new Date()
    },
    hasTime: {
      type: Boolean,
      default: !1
    },
    noSeconds: {
      type: Boolean,
      default: !1
    },
    noDays: {
      type: Boolean,
      default: !1
    },
    noYears: {
      type: Boolean,
      default: !1
    },
    noMonths: {
      type: Boolean,
      default: !1
    },
    shortMonth: {
      type: Boolean,
      default: !1
    },
    classes: {
      type: String,
      default: ""
    },
    locales: {
      type: [String, Array],
      default: navigator.languages
    }
  },
  setup(n) {
    const t = n, l = a(() => {
      let e = {};
      return e.year = "numeric", t.noDays || (e.day = "numeric"), t.noYears || (e.year = "numeric"), t.noMonth || (e.month = "numeric"), t.shortMonth ? e.month = "short" : e.month = "long", t.hasTime && (e.hour = "numeric", e.minute = "numeric", t.noSeconds || (e.second = "numeric")), e;
    }), s = a(() => {
      let e = "";
      return t.small && (e += "dateformat-small"), e;
    });
    return (e, f) => (r(), u("span", {
      class: c(o(s))
    }, i(new Date(t.date).toLocaleString(n.locales, o(l))), 3));
  }
}, d = {
  install: (n, t) => {
    n.component("DateFormat", m);
  }
};
export {
  d as default
};
